solutions/src/main/java/ru/sberbank/school/task08/KryoManager.java

```diff
@@ -0,0 +1,84 @@
+package ru.sberbank.school.task08;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import lombok.NonNull;
+import ru.sberbank.school.task08.state.GameObject;
+import ru.sberbank.school.task08.state.InstantiatableEntity;
+import ru.sberbank.school.task08.state.MapState;
+import ru.sberbank.school.util.Solution;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+@Solution(8)
+public class KryoManager extends SaveGameManager<MapState<GameObject>, GameObject> {
+
+    private Kryo kryo;
+
+    public KryoManager(@NonNull String filesDirectoryPath) {
+        super(filesDirectoryPath);
+    }
+
+    @Override
+    public void initialize() {
+        kryo = new Kryo();
+
+        kryo.register(MapState.class, new  SerializerForKryo());
+        kryo.register(InstantiatableEntity.Type.class);
+        kryo.register(InstantiatableEntity.Status.class);
+        kryo.register(GameObject.class);
+        kryo.register(ArrayList.class);
+
+    }
+
+    @Override
+    public void saveGame(String filename, MapState<GameObject> gameState) throws SaveGameException {
+        Objects.requireNonNull(filename, "Имя файла не может быть null");
+        Objects.requireNonNull(gameState, "Состояние не может быть null");
+
+        try (OutputStream os = new FileOutputStream(filesDirectory + File.separator + filename);
+            Output output = new Output(os)) {
```

Comment:
&#39;try resource&#39; has incorrect indentation level 12, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a731803260>*